src/commonMain/kotlin/fr/acinq/eclair/channel/ChannelTypes.kt

```diff
@@ -184,61 +214,69 @@ data class RemoteCommitPublished(
         // even if our txs only have one input, maybe our counterparty uses a different scheme so we need to iterate
         // over all of them to check if they are relevant
         val relevantOutpoints = tx.txIn.map { it.outPoint }.filter {
-            // is this the commit tx itself ? (we could do this outside of the loop...)
+            // is this the commit tx itself? (we could do this outside of the loop...)
             val isCommitTx = commitTx.txid == tx.txid
             // does the tx spend an output of the commitment tx?
             val spendsTheCommitTx = commitTx.txid == it.txid
             isCommitTx || spendsTheCommitTx
         }
         // then we add the relevant outpoints to the map keeping track of which txid spends which outpoint
-        return this.copy(irrevocablySpent = irrevocablySpent + relevantOutpoints.map { it to tx.txid }.toMap())
+        return this.copy(irrevocablySpent = irrevocablySpent + relevantOutpoints.map { it to tx }.toMap())
```

Comment:
```suggestion
        return this.copy(irrevocablySpent = irrevocablySpent + relevantOutpoints.associateWith { tx })
```

Reply:
Done in https://github.com/ACINQ/eclair-kmp/pull/229/commits/4b46d6121879bb9ef82542100ae63e5a1fd62ec4